tests/pytests/unit/states/test_boto_iot.py

```diff
@@ -0,0 +1,688 @@
+    """
+    if not HAS_BOTO:
+        return False
+    elif LooseVersion(boto3.__version__) < LooseVersion(required_boto3_version):
+        return False
+    elif LooseVersion(found_botocore_version) < LooseVersion(required_botocore_version):
+        return False
+    else:
+        return True
+
+
+if _has_required_boto():
+    region = "us-east-1"
+    access_key = "GKTADJGHEIQSXMKKRBJ08H"
+    secret_key = "askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs"
+    conn_parameters = {
+        "region": region,
+        "key": access_key,
+        "keyid": secret_key,
+        "profile": {},
+    }
+    error_message = (
+        "An error occurred (101) when calling the {0} operation: Test-defined error"
+    )
+    not_found_error = ClientError(
+        {
+            "Error": {
+                "Code": "ResourceNotFoundException",
+                "Message": "Test-defined error",
+            }
+        },
+        "msg",
+    )
+    topic_rule_not_found_error = ClientError(
+        {"Error": {"Code": "UnauthorizedException", "Message": "Test-defined error"}},
+        "msg",
+    )
+    error_content = {"Error": {"Code": 101, "Message": "Test-defined error"}}
+    policy_ret = dict(
+        policyName="testpolicy",
+        policyDocument='{"Version": "2012-10-17", "Statement": [{"Action": ["iot:Publish"], "Resource": ["*"], "Effect": "Allow"}]}',
+        policyArn="arn:aws:iot:us-east-1:123456:policy/my_policy",
+        policyVersionId=1,
+        defaultVersionId=1,
+    )
+    topic_rule_ret = dict(
+        ruleName="testrule",
+        sql="SELECT * FROM 'iot/test'",
+        description="topic rule description",
+        createdAt="1970-01-01",
+        actions=[{"iot": {"functionArn": "arn:aws:::function"}}],
+        ruleDisabled=True,
+    )
+    principal = "arn:aws:iot:us-east-1:1234:cert/21fc104aaaf6043f5756c1b57bda84ea8395904c43f28517799b19e4c42514"
+
+    thing_type_name = "test_thing_type"
+    thing_type_desc = "test_thing_type_desc"
+    thing_type_attr_1 = "test_thing_type_search_attr_1"
+    thing_type_ret = dict(
+        thingTypeName=thing_type_name,
+        thingTypeProperties=dict(
+            thingTypeDescription=thing_type_desc,
+            searchableAttributes=[thing_type_attr_1],
+        ),
+        thingTypeMetadata=dict(
+            deprecated=False, creationDate="2010-08-01 15:54:49.699000+00:00"
+        ),
+    )
+    deprecated_thing_type_ret = dict(
+        thingTypeName=thing_type_name,
+        thingTypeProperties=dict(
+            thingTypeDescription=thing_type_desc,
+            searchableAttributes=[thing_type_attr_1],
+        ),
+        thingTypeMetadata=dict(
+            deprecated=True,
+            creationDate="2010-08-01 15:54:49.699000+00:00",
+            deprecationDate="2010-08-02 15:54:49.699000+00:00",
+        ),
+    )
+    thing_type_arn = "test_thing_type_arn"
+    create_thing_type_ret = dict(
+        thingTypeName=thing_type_name, thingTypeArn=thing_type_arn
+    )
```

Comment:
```suggestion
class GlobalConfig:
    region = "us-east-1"
    access_key = "GKTADJGHEIQSXMKKRBJ08H"
    secret_key = "askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs"
    conn_parameters = {
        "region": region,
        "key": access_key,
        "keyid": secret_key,
        "profile": {},
    }
    error_message = (
        "An error occurred (101) when calling the {0} operation: Test-defined error"
    )
    not_found_error = ClientError(
        {
            "Error": {
                "Code": "ResourceNotFoundException",
                "Message": "Test-defined error",
            }
        },
        "msg",
    )
    topic_rule_not_found_error = ClientError(
        {"Error": {"Code": "UnauthorizedException", "Message": "Test-defined error"}},
        "msg",
    )
    error_content = {"Error": {"Code": 101, "Message": "Test-defined error"}}
    policy_ret = dict(
        policyName="testpolicy",
        policyDocument='{"Version": "2012-10-17", "Statement": [{"Action": ["iot:Publish"], "Resource": ["*"], "Effect": "Allow"}]}',
        policyArn="arn:aws:iot:us-east-1:123456:policy/my_policy",
        policyVersionId=1,
        defaultVersionId=1,
    )
    topic_rule_ret = dict(
        ruleName="testrule",
        sql="SELECT * FROM 'iot/test'",
        description="topic rule description",
        createdAt="1970-01-01",
        actions=[{"iot": {"functionArn": "arn:aws:::function"}}],
        ruleDisabled=True,
    )
    principal = "arn:aws:iot:us-east-1:1234:cert/21fc104aaaf6043f5756c1b57bda84ea8395904c43f28517799b19e4c42514"

    thing_type_name = "test_thing_type"
    thing_type_desc = "test_thing_type_desc"
    thing_type_attr_1 = "test_thing_type_search_attr_1"
    thing_type_ret = dict(
        thingTypeName=thing_type_name,
        thingTypeProperties=dict(
            thingTypeDescription=thing_type_desc,
            searchableAttributes=[thing_type_attr_1],
        ),
        thingTypeMetadata=dict(
            deprecated=False, creationDate="2010-08-01 15:54:49.699000+00:00"
        ),
    )
    deprecated_thing_type_ret = dict(
        thingTypeName=thing_type_name,
        thingTypeProperties=dict(
            thingTypeDescription=thing_type_desc,
            searchableAttributes=[thing_type_attr_1],
        ),
        thingTypeMetadata=dict(
            deprecated=True,
            creationDate="2010-08-01 15:54:49.699000+00:00",
            deprecationDate="2010-08-02 15:54:49.699000+00:00",
        ),
    )
    thing_type_arn = "test_thing_type_arn"
    create_thing_type_ret = dict(
        thingTypeName=thing_type_name, thingTypeArn=thing_type_arn
    )
```
Worst case, it should be like the above.
Ideally, the attribute setup should be deferred, ie:
```python
class GlobalConfig:
    def __init__(self):
        self.region = "us-east-1"
        ... etc ...
```

Then return `GlobalConfig()` in a fixture and use that fixture.